### ISSUE TITLE
Bedrock clients cannot merge enchant books

### ISSUE THREAD

**Issue by sudokou** [GC-spigot/AdvancedEnchantments#3578]
### Describe the bug

Bedrock players are unable to use an anvil to merge 2 enchant books together (in order to increase the enchant's level). Drag & drop in inventory doesn't work either.

Java players are able to use anvil.

### How to reproduce

Attempt to merge 2 books of the same enchant on bedrock.

### "/ae plinfo" link

https://paste.md-5.net/ipaholicuh

**Comment by KNzE07** [comment]
Why can't your server merge enchant books? My server can access both Java and Bedrock, but they can.

**Comment by sudokou** [comment]
> Why can't your server merge enchant books? My server can access both Java and Bedrock, but they can.

How do your bedrock players do it? In an anvil? Also, to clarify, you are saying that bedrock players can merge _custom_ enchant books, correct?

**Comment by DimaSergeew** [comment]
> ### Опишите ошибку
> Игроки Bedrock не могут использовать наковальню для объединения 2 зачарованных книг вместе (чтобы повысить уровень зачарования). Перетаскивание в инвентаре также не работает.
> 
> Java-плееры могут использовать anvil.
> 
> ### Как воспроизвести
> Попытайтесь объединить 2 книги с одним и тем же зачарованием на bedrock.
> 
> ### Скриншоты / Видео
> _Ответа нет_
> 
> ### ссылка "/ ae plinfo"
> https://paste.md-5.net/ipaholicuh
> 
> ### Журнал сервера
> _Ответа нет_

My Bedrock players are doing this without any problems; what's going on?

**Comment by sudokou** [comment]
> My Bedrock players are doing this without any problems; what's going on?

Again, How do your bedrock players do it? In an anvil? Also, to clarify, you are saying that bedrock players can merge custom enchant books, correct?

**Comment by ThomasWega** [comment]
@sudokou Have you figured out a solution? It seems like other people don't have this issue

**Comment by sudokou** [comment]
> @sudokou Have you figured out a solution? It seems like other people don't have this issue

No, I've validated the issue myself and checked every AE config for options related to "merging", "drag" (n drop), and "anvil".

Nobody that responded has confirmed that they understood my issue correctly or given me any instruction on how to replicate their "success". Without specific steps to follow from the random replies, I cannot confirm whether the issue is isolated to myself or not. I am wondering if the other responses were testing merging vanilla enchant books - as that, ofc, works.

**Comment by aXe18** [comment]
@sudokou First of all, its java plugin, so bedrock is not supported, and thats why some thins may not work like "they should be"
But drag&drop feature is not working on bedrock (or at least i dont know way), you can merge books in anvil with bedrock client.
Thats how i have heard its done, but i have 0 experience about it myself

**Comment by sudokou** [comment]
To clarify, drag-n-drop works on bedrock for applying custom enchant books to items. It does not work on bedrock, or Java, for merging custom enchant books - this is not currently a feature of the plugin. **One solution to my ticket would be to add this as a feature.**

Alternatively, anvil support should be added to bedrock clients. From what I can recall, anvils do not work at all with custom enchants for a bedrock player - bedrock players cannot apply custom enchants to items or merge custom enchant books together. (I'd have to retest and validate whether applying to items works or not - it did not a year ago and we just switched to the drag-n-drop method).

My initial post was only reporting that merging books doesn't work because there is already a workaround for the other anvil function (applying to items: bedrock players can use drag and drop). Therefore, the only functionality they are entirely unable to use is merging custom enchant books together.
